src/ORM/Association.php

```diff
@@ -836,17 +836,19 @@ public function defaultRowValue(array $row, bool $joined): array
      *
      * @param array<string, mixed>|string|null $type the type of query to perform, if an array is passed,
      *   it will be interpreted as the `$options` parameter
-     * @param array<string, mixed> $options The options to for the find
+     * @param mixed ...$args Arguments that match up to finder-specific parameters
      * @see \Cake\ORM\Table::find()
      * @return \Cake\ORM\Query\SelectQuery
      */
-    public function find(array|string|null $type = null, array $options = []): SelectQuery
+    public function find(array|string|null $type = null, mixed ...$args): SelectQuery
     {
```

Comment:
```suggestion
    public function find(array|string|null $type = null, mixed $args): SelectQuery
    {
        if (count($args) == 1 && is_array($args[0])) {
            $args = $args[0];
        }
```

This could let us handle backwards compatibility better. If we need it in multiple places we could have a protected method for this logic.

Reply:
Always assuming that a single array argument is an options array would mean one wouldn't be able to make a new finder which has an array arg which isn't options. For eg. `findWithIds(SelectQuery $select, array $ids)`.

Reply:
Yes. That is a limitation of the backwards compatibility I proposed, a work around for this limitation would be to call that finder like `find('translations', $locales=['en'])` Because we got named parameters as the input we can call the finder with named parameters. We could either use reflection to detect how to call the finder or infer it based on the parameter usage. This helps incentivize upgrading which could be a nice win.

Reply:
I can take a pass at implementing this logic.

Reply:
The changed you proposed isn't necessary. The check for old style finder with $options is already done in `Table::invokeFinder()`. There's also a test case `AssociationTest::testCustomFinderWithOptions()` which tests for call to `Assocation::find()` which calls `TestTable::findWithOptions()` which has the `$options` argument.

Plus the logic I currently have still allows `find('translations', ['en'])` usage since when reflecting I check for the argument name to decide if it's the old $options argument.

Please feel free to add more tests for `Table::find()` and `Association::find()` for BC usage compatibility.